src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/AbstractFactory.php

```diff
@@ -47,7 +47,7 @@ abstract class AbstractFactory implements SecurityFactoryInterface
         'failure_path_parameter' => '_failure_path',
     ];
 
-    public function create(ContainerBuilder $container, $id, $config, $userProviderId, $defaultEntryPointId)
+    public function create(ContainerBuilder $container, string $id, array $config, ?string $userProviderId, ?string $defaultEntryPointId)
```

Comment:
`$userProviderId` is nullable but it is passed to `createAuthProvider` and `createListener` where it's not nullable.

Reply:
good catch, thanks